tools/pot/openvino/tools/pot/algorithms/quantization/accuracy_aware_gna/algorithm.py

```diff
@@ -61,44 +71,126 @@ def _request_alt_statistics(self, model):
+        unify_conv_activation_weight = self._config['unify_conv_activation_weight']
+
+        target_nodes = []
+        # Since GNA3.0 support convolution layer of 8/8 or 16/16 (activation/weight),
+        # we should convert them simultaneously
+        if unify_conv_activation_weight:
+            for fq_name in nodes_names:
+                fq = mu.get_node_by_name(model, fq_name)
+                fq_input = nu.get_node_input(fq, 0)
+                if fq_input.type != 'Const':
+                    continue
+                if fq.levels > 2 ** self._default_fq_bit:
+                    logger.debug('Skipping %s because it\'s already in %s precision (levels: %d)',
+                            fq_name, self._precision_change_to, fq.levels)
+                    continue
 
-        self._oc_stats = self._stats_collector.get_statistics_for_algorithm(self._oc_algo.name)
+                output_node = nu.get_node_output(fq, 0)[0]
 
-        weight_fq_names = []
-        for fq_name in nodes_names:
-            fq = mu.get_node_by_name(model, fq_name)
-            fq_input = nu.get_node_input(fq, 0)
-            if fq_input.type != 'Const':
-                logger.debug('Skipping %s because it\'s not weight FQ', fq_name)
-                continue
-            if fq.levels > 2 ** self._default_fq_bit:
-                logger.debug('Skipping %s because it\'s already in %s precision (levels: %d)',
-                             fq_name, self._precision_change_to, fq.levels)
-                continue
-            weighted_node = nu.get_node_output(fq, 0)[0]
-            node_quantization_config = {
-                fq_name: self._acc_fq_config[fq_name],
-                'fqs_to_rescale': []
-            }
-            logger.debug('Change bits for %s', fq_name)
-            self._minmax_algo.fill_fq_range(model, self._acc_weights_stats, [], node_quantization_config, self._config)
-            fq.levels = compute_levels(self._acc_fq_config[fq_name], True)
-
-            if self._oc_stats:
-                bias_node = nu.get_bias_for_node(weighted_node)
-                if bias_node is None:
+                # When fq for weight is not belong to convolution,
+                # you may convert it independentely
+                if output_node.type != 'Convolution':
+                    target_nodes.append((None, fq))
                     continue
-                add_node_name = nu.get_node_output(bias_node, 0)[0].name
-                if add_node_name not in self._oc_stats \
-                        or 'max_per_tensor' not in self._oc_stats[add_node_name]:
+
+                fq_activation = nu.get_node_input(output_node, 0)
+
+                if fq_activation.type != 'FakeQuantize':
+                    logger.debug('Skipping %s because it\'s not FQ for conv activation', fq_activation.fullname)
+                    continue
+
+                target_nodes.append((fq_activation, fq))
+        else:
```

Comment:
Maybe we can somehow collapse these code duplications?

Reply:
I refined the logic to minimize code duplications.